test/websockex_test.exs

```diff
@@ -219,17 +220,17 @@ defmodule WebSockexTest do
   end
 
   test "handles a tcp message send right after connecting", context do
-    send(context.server_pid, :immediate_reply)
+    TestServer.new_conn_mode(context.server_pid, :immediate_reply)
```

Comment:
Here we have to block until the mode switching is completed, otherwise, there will be occasional failures due to parallel messaging.